pkg/storage/storage.go

```diff
@@ -187,13 +190,29 @@ type ChangelogBackend interface {
 	ReadChanges(ctx context.Context, store, objectType string, paginationOptions PaginationOptions, horizonOffset time.Duration) ([]*openfgapb.TupleChange, []byte, error)
 }
 
+// TimeProvider has functionality of Now that allows mocking of current time
+type TimeProvider interface {
+	Now() time.Time
+}
+
+// DefaultTimeProvider allows returning current time as Now()
+type DefaultTimeProvider struct{}
```

Comment:
I suggest we do the following:

```go
package storage

// imports...

var DefaultTimeProvider = &defaultTimeProvider{}

... // other code

type defaultTimeProvider struct {}

var _ TimeProvider = (*defaultTimeProvider)(nil) // ensures we always implement the TimeProvider interface

func (d *defaultTimeProvider) Now() time.Time {
    return time.Now()
}
```

then usage looks like:

```go
var ds OpenFGADatastore

ds.SetTimeProvider(DefaultTimeProvider) // more idiomatic than ds.SetTimeProvider(DefaultTimeProvider{})
```

Reply:
Obsolete as we don't have the time provider anymore